week3/average/maverickchaser/Main.hs

```diff
@@ -0,0 +1,11 @@
+average a = sum a / fromIntegral (length a)
+
+-- Method 1
+
+
+moving k a = premoving (take (k-1) a) 0 1 ++ movingImpl 0
+    where premoving  [] _ _ = []
+          premoving (x:a) cur index = (x+cur) / fromIntegral index : premoving a (cur+x) (index+1)
+          movingImpl index
```

Comment:
If you passed the list as well as the index, you could get rid of `drop index a`. Also, calculating `length a` every time is wasteful (`length` is O(n))